Firestore/core/src/model/value_util.h

```diff
@@ -58,14 +63,69 @@ util::ComparisonResult Compare(const google_firestore_v1_Value& left,
 bool Equals(const google_firestore_v1_Value& left,
             const google_firestore_v1_Value& right);
 
+bool Equals(const google_firestore_v1_ArrayValue& left,
+            const google_firestore_v1_ArrayValue& right);
+
 /**
  * Generate the canonical ID for the provided field value (as used in Target
  * serialization).
  */
 std::string CanonicalId(const google_firestore_v1_Value& value);
 
+/**
+ * Generate the canonical ID for the provided array value (as used in Target
+ * serialization).
+ */
+std::string CanonicalId(const google_firestore_v1_ArrayValue& value);
+
+/** Returns true if the array value contains the specified element. */
+bool Contains(google_firestore_v1_ArrayValue haystack,
+              google_firestore_v1_Value needle);
+
+/** Returns `nullptr` in its Protobuf representation. */
```

Comment:
Maybe `Returns a null Protobuf value`? `nullptr` is pretty specifically a pointer to address 0.

Reply:
Fixed